src/consensus/encode.rs

```diff
@@ -622,34 +626,23 @@ impl Decodable for Vec<u8> {
         if len > MAX_VEC_SIZE {
             return Err(self::Error::OversizedVectorAllocation { requested: len, max: MAX_VEC_SIZE })
         }
-        let mut ret = Vec::with_capacity(len);
-        ret.resize(len, 0);
+        let mut ret = vec![0u8; len];
         d.read_slice(&mut ret)?;
         Ok(ret)
     }
 }
 
 impl Encodable for Box<[u8]> {
     #[inline]
-    fn consensus_encode<S: io::Write>(&self, mut s: S) -> Result<usize, Error> {
-        let vi_len = VarInt(self.len() as u64).consensus_encode(&mut s)?;
-        s.emit_slice(&self)?;
-        Ok(vi_len + self.len())
+    fn consensus_encode<S: io::Write>(&self, s: S) -> Result<usize, Error> {
+        consensus_encode_with_size(self, s)
     }
 }
 
 impl Decodable for Box<[u8]> {
     #[inline]
     fn consensus_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
```

Comment:
Can you fix this commit to remove the now-unused `mut`?

Reply:
Done